### DISCUSSION TITLE
How to update component state from outside (V3.x)

### DISCUSSION THREAD

**Discussion by jmariller** [alpinejs/alpine#2569] [1. Help] [answered]
Hi,

I know there is already an issue that covered this topic but it was for version 2.x (see #426), where the `__x` property could be used to interact with data from outside Alpine - now I am trying to do the same with version 3.x, which does not work anymore. I understand we could use custom events, but I find this a bit tricky when it comes to showing for example the list of existing data in a dynamic way.

Any advices/ideas?

Many thanks!

**Comment by SimoTod** [comment, marked as answer]
You can use an Alpine store for the property you want to share externally

**Reply by jmariller** [reply]
Thank you, that works!
